minio/helpers.py

```diff
@@ -109,6 +152,10 @@ def get_target_url(endpoint, bucket_name=None, object_name=None, query=None):
 
     return ''.join(url_components)
 
+allowed_hostname = re.compile(
+    "^((?!-)[A-Z\\d-]{1,63}(?<!-)\\.)*((?!-)[A-Z\\d-]{1,63}(?<!-))$",
+    re.IGNORECASE)
```

Comment:
Sorry, I couldn't able to notice that.  You could move globals into beginning after the imports and rename then like _ALLOWED_HOSTNAME_REGEX.